src/main/kotlin/no/nav/medlemskap/regler/v1/frilanser/ErArbeidsforholdetFrilanserRegel.kt

```diff
@@ -0,0 +1,46 @@
+package no.nav.medlemskap.regler.v1.frilanser
+
+import no.nav.medlemskap.domene.Datagrunnlag
+import no.nav.medlemskap.domene.InputPeriode
+import no.nav.medlemskap.domene.Ytelse
+import no.nav.medlemskap.domene.arbeidsforhold.Arbeidsforhold
+import no.nav.medlemskap.domene.arbeidsforhold.Arbeidsforhold.Companion.arbeidsforholdForYrkestype
+import no.nav.medlemskap.domene.arbeidsforhold.Arbeidsforholdstype
+import no.nav.medlemskap.regler.common.Funksjoner.alleEr
+import no.nav.medlemskap.regler.common.RegelId
+import no.nav.medlemskap.regler.common.Resultat
+import no.nav.medlemskap.regler.common.Resultat.Companion.ja
+import no.nav.medlemskap.regler.common.Resultat.Companion.nei
+import no.nav.medlemskap.regler.v1.arbeidsforhold.ArbeidsforholdRegel
+import java.time.LocalDate
+
+class ErArbeidsforholdetFrilanserRegel(
+    ytelse: Ytelse,
+    private val periode: InputPeriode,
+    private val førsteDagForYtelse: LocalDate?,
+    private val arbeidsforhold: List<Arbeidsforhold>,
+    regelId: RegelId = RegelId.REGEL_17
+) : ArbeidsforholdRegel(regelId, ytelse, periode, førsteDagForYtelse) {
+
+    override fun operasjon(): Resultat {
+        return when {
+            arbeidsforhold.isEmpty() -> nei(regelId)
+            arbeidsforhold.arbeidsforholdForYrkestype(kontrollPeriodeForArbeidsforhold) alleEr Arbeidsforholdstype.FRILANSER -> ja(
```

Comment:
er det funksjonelle avklart på om alle arbeidsforhold må være frilanser for å få Ja på denne regelen? Hva om en bruker har hatt ordinært arbeidsforhold i en del av kontrollperioden, og frilanser i andre halve, er det nok til at en person regnes som vanlig arbeidstaker?

Reply:
Det må nesten tas opp med Helle, siden det kan jeg ikke svare på.

Vil tro at så lenge det bare er ett arbeidsforhold som er frilanser så går det til uavklart

Reply:
Det kan vi gjøre. Enig, er nok antakelig at så lenge det bare er ett arbeidsforhold som er frilanser så skal det gå til uavklart. I såfall må da "alleEr" erstattes med noe i retning "finnes"